### DISCUSSION TITLE
Is there an SDK equivalent for the 'fga tuple read' command?

### DISCUSSION THREAD

**Discussion by keitherw** [openfga/community#387] [Q&A] [answered]
I've tried to browse the code that performs the 'fga tuple read' but cannot find it. I'd like an SDK request that would provide all the tuples in JSON format just like the cli.

**Comment by adriantam** [comment, marked as answer]
Hello @keitherw , the read is a [native API ](https://openfga.dev/api/service#/Relationship%20Tuples/Read) that is supported by all SDKs.  For example, in Go SDK, you can call [Read](https://github.com/openfga/go-sdk?tab=readme-ov-file#read-relationship-tuples) from which you can [marshal](https://pkg.go.dev/encoding/json) the data in JSON format.

**Reply by keitherw** [reply]
Awesome, thanks. That worked.

	// Read all stored relationship tuples
	body := client.ClientReadRequest{}

	options := client.ClientReadOptions{
		PageSize: openfga.PtrInt32(10),
	}

	data, err := fgaClient.Read(context.Background()).Body(body).Options(options).Execute()
	if err != nil {
		log.Println(err)
		http.Error(
			w,
			err.Error(),
			http.StatusInternalServerError)
		return
	}
